dash_core_components/_graph.py

```diff
@@ -0,0 +1,141 @@
+            "unhovers" from a point. If False, then
+            the `hoverData` property will be equal to
+            the data from the last point that was
+            hovered over.(default=false)
+        :param selectedData: Data from latest select event. Read-
+            only.(default=null)
+        :param relayoutData: Data from latest relayout event which occurs
+            when the user zooms or pans on the plot or
+            other layout-level edits. Has the form `{<attr
+            string>: <value>}` describing the changes
+            made. Read-only.(default=null)
+        :param extendData: Data that should be appended to existing traces.
+            Has the form `[updateData, traceIndices,
+            maxPoints]`, where `updateData` is an object
+            containing the data to extend, `traceIndices`
+            (optional) is an array of trace indices that
+            should be extended, and `maxPoints` (optional)
+            is either an integer defining the maximum number
+            of points allowed or an object with key:value
+            pairs matching `updateData` Reference the
+            Plotly.extendTraces API for full usage:
+            https://plot.ly/javascript/plotlyjs-function-
+            reference/#plotlyextendtraces(default=null)
+        :param restyleData: Data from latest restyle event which occurs
+            when the user toggles a legend item, changes
+            parcoords selections, or other trace-level
+            edits. Has the form `[edits, indices]`, where
+            `edits` is an object `{<attr string>: <value>}`
+            describing the changes made, and `indices` is
+            an array of trace indices that were edited.
+            Read-only.(default=null)
+        :param figure: Plotly `figure` object. See schema:
+            https://plot.ly/javascript/reference Only supports
+            `data` array and `layout` object. `config` is set
+            separately by the `config` property, and `frames` is
+            not supported.(default={data: [], layout: {}})
+        :param style: Generic style overrides on the plot div
+        :param className: className of the parent div
+        :param animate: Beta: If true, animate between updates using
+            plotly.js's `animate` function(default=false)
+        :param animation_options: Beta: Object containing animation
+            settings. Only applies if `animate` is
+            `true`(default={     frame: {
+            redraw: false,     },     transition: {
+            duration: 750,         ease: 'cubic-in-
+            out',     }, })
+        :param config: Plotly.js config options. See
+            https://plot.ly/javascript/configuration-options/
+            for more info.(default={     staticPlot: false,
+            editable: false,     edits: {
+            annotationPosition: false,         annotationTail:
+            false,         annotationText: false,
```

Comment:
Hey re: https://github.com/plotly/dash-core-components/issues/514 I'm thinking of removing these defaults, just using `config: {}`, as well as writing some sort of test that the `PropTypes.shape` we give matches the available options in the current version of plotly.js. But in this PR the default seems to be the only info about what's allowed inside `config` that makes it into the Python docs. How do you think we should handle this?
- bring the shape into this docstring programmatically?
- require that this shape be described explicitly in the docstring for `graphPropTypes.config` (and then I'd write a test to verify that this is kept up-to-date)?
- something else?

Reply:
The default in the docstring come from `Graph.defaultProps`, if you change it to `{}` they will be removed.

> bring the shape into this docstring programmatically?

You cannot do that on at runtime with react-docgen as nothing is evaluated just parsed. It has to be in the source. Maybe a script that fetch and format and you insert the output in the docstring, I'd imagine you'd only have to do that when you update plotly.js.

> require that this shape be described explicitly in the docstring for graphPropTypes.config (and then I'd write a test to verify that this is kept up-to-date)?

I like that.

> something else?

Front-end validations and keep as is with the default `{}`?

Reply:
Come to think of it, the mypy validations for `shapes` is shallow and not key based, it might be worthwhile to format the PropType definition in the docstring.

Reply:
OK cool - one way or another we can handle this in documentation. For now I'm going to take the defaults out for https://github.com/plotly/dash-core-components/issues/514